packages/angular/projects/clr-angular/src/forms/datepicker/date-container.ts

```diff
@@ -169,9 +164,9 @@ export class ClrDateContainer implements DynamicWrapper, OnDestroy, AfterViewIni
     this.subscriptions.push(
       this.ifControlStateService.statusChanges.subscribe((state: CONTROL_STATE) => {
         this.state = state;
-        this.valid = CONTROL_STATE.VALID === state;
+        this.valid = CONTROL_STATE.VALID === state && !!this.controlSuccessComponent;
```

Comment:
I wonder why we haven't extended `ClrAbstractContainer` in this `ClrDateContainer`.

Reply:
Because it was written long before and it was a major refactoring. I see us fixing this with Core and not Angular.

Reply:
I manage to rework it so date-container to extend `ClrAbstractContainer` but have some hard time making one test to pass .. :)